spec-tests/src/test/java/org/candlepin/spec/jobs/JobStatusSpecTest.java

```diff
@@ -0,0 +1,498 @@
+/**
+ * Copyright (c) 2009 - 2022 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.spec.jobs;
+
+import static java.lang.Thread.sleep;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import org.candlepin.ApiException;
+import org.candlepin.dto.api.v1.AsyncJobStatusDTO;
+import org.candlepin.dto.api.v1.ConsumerDTO;
+import org.candlepin.dto.api.v1.OwnerDTO;
+import org.candlepin.dto.api.v1.PoolDTO;
+import org.candlepin.dto.api.v1.ProductDTO;
+import org.candlepin.dto.api.v1.UserDTO;
+import org.candlepin.resource.ConsumerApi;
+import org.candlepin.resource.JobsApi;
+import org.candlepin.resource.OwnerApi;
+import org.candlepin.resource.OwnerProductApi;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+import org.candlepin.spec.bootstrap.client.ApiClients;
+import org.candlepin.spec.bootstrap.client.JobsClient;
+import org.candlepin.spec.bootstrap.client.SpecTest;
+import org.candlepin.spec.bootstrap.client.SpecTestFixture;
+import org.candlepin.spec.bootstrap.data.builder.Consumers;
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+import org.candlepin.spec.bootstrap.data.util.StringUtil;
+import org.candlepin.spec.bootstrap.data.util.UserUtil;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+
+/**
+ * Test the /status resource
+ */
+@SpecTest
+class JobStatusSpecTest extends SpecTestFixture {
+
+    private static ApiClient client;
+    ApiClient userClient;
+    private static OwnerDTO owner;
+    private static UserDTO user;
+    private OwnerApi ownerApi;
+    private OwnerProductApi ownerProductApi;
+    private ConsumerApi consumerApi;
+    private JobsClient jobsClient;
+    private ProductDTO product;
+    private PoolDTO pool;
+
+
+    @BeforeEach
+    public void beforeEach() throws ApiException {
```

Comment:
We should setup clients per test class. There's no need to recreated them for each test.

Reply:
Method is renamed, but the annotations is still @BeforeEach